### PR TITLE
New: Handle select groups that do not follow -RlsGrp format

### LINKED ISSUE #3550: add support for {ReleaseGroup} to detect QxR encoders in movie naming
**Is your feature request related to a problem? Please describe.**
Most of QxR encoders (Tigole, FreetheFish, Silence, afm72, etc) add their name without the "-" so it doesnt get parsed as a {ReleaseGroup} resulting in a "file name (radarr)" instead of "file name (Tigole)" for example.

Im asking for support on this issue because I think QxrR are becoming the main x265 content encoders.

**Describe the solution you'd like**
Id like to have the encoder properly added as a {ReleaseGroup}. Maybe for started add a list of known encoders that do this and if radarr detects them, add them to the filename.

**Describe alternatives you've considered**
An even better option would be leaving this to the user choice, maybe by implementing some sort of conditional naming option. For example let users have the option to add certain different things to the filename if some conditions are met.

Sorry if this has been discussed before or if what im asking is too much to implement, Im just trying to suggest something i think would be usefull for the project.

Thank you

### LINKED ISSUE #5804: Incorrectly Parsing IMDB/TMDB as Group
**Describe the bug**

For manual imports of movies organized like:
`/data/media/movies-archive/Lock Stock and Two Smoking Barrels (1998) {imdb-tt0120735}/Lock Stock and Two Smoking Barrels (1998) [imdb-tt0120735][Bluray-1080p][8bit][x264][DTS-HD MA 5.1]-FraMeSToR.mkv`

Radarr is grabbing the imdb ID as the group rather than the actual group.

Imported To
/data/media/movies/Lock Stock and Two Smoking Barrels (1998) {imdb-tt0120735}/Lock Stock and Two Smoking Barrels (1998) [imdb-tt0120735][Bluray-1080p][8bit][x264][DTS 5.1]-tt0120735.mkv

**To Reproduce**

Manual import a file within a folder where both have the imdb (or presumably tmdb) id

**Expected behavior**

Correct group is parsed
**Screenshots**

N/A
**Platform Information (please complete the following information):**
 - OS:  Agnostic
 - Docker:  Agnostic 
 - Mono or .NET Version (System -> Status):  Agnostic
 - Browser and Version (Only needed for UI issues):  N/A
 - Radarr Version:  3.0.2.4460
 - Radarr Branch:  nightly

**Trace Logs**
Turn on Trace logs under Settings -> General and wait for the bug to occur again. 
**Upload the full log file here (or another site (e.g. pastebin) and link it). Issues will be closed, if they do not include this!**
Too old to have; likely can reproduce if needed

### PR DESCRIPTION
**Pull request by bakerboy448**
#### Database Migration
NO

#### Description
Handle a handful of groups who don't follow standard naming.
These are manual exceptions and any additions will need to be manually added.

Currently this will handle the following groups that follow the format of `GroupName)` or `GroupName]`

- Tigole
- Joy
- YIFY
- YTS.MX
- FreetheFish
- afm72

It will also handle the following groups by checking for a Case Insensitive match against the exact name
- KRaLiMaRKo
- E.N.D

For future requests, a GHI should be opened with the release name and I should be pinged.

#### Todos
- [X] Tests
- [x] Translation Keys N/A
- [x] Wiki Updates N/A

#### Issues Fixed or Closed by this PR

* Fixes #3550
* Fixes #5804 (see further comments)

### CODE REVIEW

**Comment by Qstick** [comment]
This doesn't look any different than the other just pushed. did you lose something in the rebase/push?

**Comment by bakerboy448** [comment]
> This doesn't look any different than the other just pushed. did you lose something in the rebase/push?

It was lost, but now its found.

**Comment by bakerboy448** [comment]
changing to WIP pending additional work on test cases for TMDB/IMDB group names (#5804)

To Dos:
- [ ] test anti TMDb/IMDb ID logic
- [ ] Create test case to reproduce #5804 and ensure we are guarded against it
@Qstick may need some help on that test case.

File: src/NzbDrone.Core/Parser/Parser.cs [hunk 1]
```diff
@@ -125,6 +125,11 @@ public static class Parser
         private static readonly Regex YearInTitleRegex = new Regex(@"^(?<title>.+?)(?:\W|_)?(?<year>\d{4})",
                                                                 RegexOptions.IgnoreCase | RegexOptions.Compiled);
 
+        //Handle Exception Release Groups that don't follow -RlsGrp; Manual List
+        //First Group is groups whose releases end with RlsGroup) or RlsGroup]  second group (entries after `(?=\]|\))|`) is name only...BE VERY CAREFUL WITH THIS, HIGH CHANCE OF FALSE POSITIVES
+        private static readonly Regex ExceptionReleaseGroupRegex = new Regex(@"(?<releasegroup>(Tigole|Joy|YIFY|YTS.MX|FreetheFish|afm72)(?=\]|\))|KRaLiMaRKo|E\.N\.D)",
```
Comment (Qstick) [inline src/NzbDrone.Core/Parser/Parser.cs:130]: Can you add some false positive tests for `joy` at least, to make sure it doesn't pickup when we don't want it to if there is no release group.

Comment (bakerboy448) [reply src/NzbDrone.Core/Parser/Parser.cs:130]: added, passed locally, pushed

**Review by Qstick — DISMISSED** [review]

**Review by Qstick — APPROVED** [review]

**Review by Qstick — APPROVED** [review]

**Pull request merged**